### PR TITLE
Adding docker_login

### PR DESCRIPTION
**Pull request by chouseknecht**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

docker_login.py 
##### ANSIBLE VERSION

```
ansible 2.1.0 (docker_testing 9bb069f873) last updated 2016/04/18 12:36:03 (GMT -400)
  lib/ansible/modules/core: (detached HEAD cf01087a30) last updated 2016/04/05 16:31:53 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 204b4bab56) last updated 2016/04/05 16:31:56 (GMT -400)
  config file =
  configured module search path = Default w/o overrides
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @chouseknecht for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/docker/docker_login.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+version_added: "2.0"
+
+description:
+    - Provides functionality similar to the "docker login" command.
+    - Authenticate with a docker registry and add the credentials to your local Docker config file. Adding the
+      credentials to the config files allows future connections to the registry using tools such as Ansible's Docker
+      modules, the Docker CLI and docker-py without needing to provide credentials.
+    - Running in check mode will perform the authentication without updating the config file.
+
+options:
```
Comment (bcoca) [inline cloud/docker/docker_login.py:39]: what is the equivalent of 'docker_url' from the original module?

Comment (chouseknecht) [inline cloud/docker/docker_login.py:39]: Not sure what you're after. In the origin module docker_url was defined as:

```
docker_url:
    description:
       - Refers to the protocol+hostname+port where the Docker server is hosted
    required: false
    default: unix://var/run/docker.sock
```

In the new Docker modules the equivalent is `docker_host` (with an alias of 'docker_url') which is defined in docker_common.py

Comment (bcoca) [inline cloud/docker/docker_login.py:39]: ah, OK, then, its not documented in the module doc fragment, but i do see the alias there.

**Pull request merged**